cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -89,8 +122,9 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 
 	start_time := time.Now()
 
-	osbuildOutput, err := RunOSBuild(args.Manifest, impl.Store, outputDirectory, os.Stderr)
+	osbuildOutput, err = RunOSBuild(args.Manifest, impl.Store, outputDirectory, os.Stderr)
 	if err != nil {
+		r = append(r, err)
```

Comment:
I'm a little worried about appending the error from osbuild to the list of target errors. Target errors have never been strongly defined but the idea is that they should be somewhat connected with upload targets.

Btw once upon a time, there was an idea to distinguish between a misbehaving worker and a failed job.

When building a particular manifest fails or an upload fails, this is a failed job that results in a failed compose.

When some other stuff fails (like creating a temporary directory on the worker or running osbuild for different reasons than a bad manifest), the worker should be treated like a misbehaving and the job should be rescheduled. We never really finished this part.

So yeah, I think this is an area where definitely is a lot of room for improvements and discussions. Overall, I think that your workaround makes the situation slightly better, I'm just worried that we're abusing `targetErrors` for something they were never intended for.

Reply:
Yes, this is exactly what I was confused about. There is no obvious place to report different kinds of errors. Let's remove this particular line and keep the rest. We can follow up later with bigger refactoring and extending the Result structure.